common/table.js

```diff
@@ -56,15 +56,54 @@
      *      });
      */
     .factory('recordTableUtils', ['DataUtils', '$timeout', function(DataUtils, $timeout) {
-        function read(scope) {
+
+        // This method sets backgroundSearch states depending upon various parameters
```

Comment:
You have not defined a 'background' vs 'foreground' search. (I can know from earlier conversations.) These need to be documented in here to explain what that means and what you are doing. Even a few sentences in a comment is good.